src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,73 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let filter_non_operands components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl -> if String.trim hd = "spc" then tl else aux (hd :: acc) tl
+  in
+  aux [] components
+
+let extract_operands k regex components =
+  let rec aux acc = function
+    | [] -> List.rev acc
+    | hd :: tl ->
+        if Str.string_match regex hd 0 then (
```

Comment:
Since this is the only use of regex, you could just put it's full definition inline here.